app/api/v1/reports/views.py

```diff
@@ -25,67 +20,67 @@ def post(self):
         parser.add_argument('type',
             choices=("Red-Flag, Intervention"),
             type=str, location="json", required=True,
-            help='Report type can only be either Red-Flag or Intervention.'
+            help="Report type can only be strictly either 'Red-Flag' or 'Intervention'."
         )
         parser.add_argument('location', type=str, location="json", help='Location cannot be blank.', required=True)
         parser.add_argument('comment', type=str, location="json", help='Comment cannot be blank.', required=True)
         parser.add_argument('status', type=str, location="json", help='Status cannot be blank.', required=True)
-        args = parser.parse_args()
-
-        data = request.get_json()
+        data = parser.parse_args()
         report = {
-            "id": self.reports.total_reports_created,
+            "id": ReportsModel.total_reports_created,
             "createdOn": datetime.datetime.now().strftime("%d-%m-%Y %H:%M"),
-            "createdBy": current_user,
+            "reporter": current_user,
             "type": data["type"],
             "location": data["location"],
             "status": data["status"],
             "comment": data["comment"],
         }
-        self.reports.save(report)
-        return {"status": 201, "data": [{"id": report["id"], "message": "Created report."}]}, 201
+        if data["type"] == ("Red-Flag" or "Intervention"):
```

Comment:
Since you are using parser, you can leverage on the choices option so that only these two choices can be expected.